### PR TITLE
update components name and organizationForm

### PR DESCRIPTION
**Pull request by Maxstgt**

### CODE REVIEW

File: libs/organization/src/lib/+state/organization.service.ts [hunk 1]
```diff
@@ -123,12 +123,12 @@ export class OrganizationService {
 
   // TODO(#679): ssmehow the updateActiveMembers array don't filter correctly 
   // the id out of the activeMembersArray.
-  public async deleteActiveSigner(member: OrganizationMember, action: OrganizationAction) {
+  /* public async deleteActiveSigner(member: OrganizationMember) {
```
Comment (GrandSchtroumpf) [inline libs/organization/src/lib/+state/organization.service.ts]: I think you can remove it now.

File: libs/organization/src/lib/+state/organization.store.ts [hunk 2]
```diff
@@ -1,12 +1,13 @@
 import { Injectable } from '@angular/core';
 import { Store, StoreConfig } from '@datorama/akita';
-import { Organization, OrganizationForm, createOrganization } from './organization.model';
+import { Organization, createOrganization } from './organization.model';
 
 export interface OrganizationState {
   org: Organization;
-  form: OrganizationForm;
+  form: any;
```
Comment (GrandSchtroumpf) [inline libs/organization/src/lib/+state/organization.store.ts]: Checkout where it's used to see if you can remove it.

File: libs/organization/src/lib/+state/organization.store.ts [hunk 3]
```diff
@@ -1,12 +1,13 @@
...
 }
 
+// TODO #687
```
Comment (GrandSchtroumpf) [inline libs/organization/src/lib/+state/organization.store.ts]: Add a comment on the todo so we know what this issue is about.

File: libs/organization/src/lib/components/organization-action-item/organization-action-item.component.ts [hunk 4]
```diff
@@ -21,7 +21,7 @@ export class OrganizationActionItemComponent implements OnInit {
 
   public dataSource;
 
-  // TODO #638: replace any and make it an Observable
+  // TODO #638: replace any
   @Input() actionItems: any;
```
Comment (GrandSchtroumpf) [inline libs/organization/src/lib/components/organization-action-item/organization-action-item.component.ts]: Why you cannot replace now ?

File: libs/organization/src/lib/components/organization-quorum-form/organization-quorum-form.component.html [hunk 5]
```diff
@@ -0,0 +1,22 @@
...
+  <mat-label>How much members needed</mat-label>
+  <mat-select>
+    <mat-option *ngFor="let number of " [value]="number">{{number}}</mat-option>
+  </mat-select>
+</mat-form-field>
+<mat-divider></mat-divider>
+<article>
+  <h2>Members</h2>
+  <form [formGroup]="control">
+    <span *ngFor="let member of this.controlContainer.control.get('activeMembers').value" fxLayout="row"
```
Comment (GrandSchtroumpf) [inline libs/organization/src/lib/components/organization-quorum-form/organization-quorum-form.component.html]: Create a `<ng-container>` for the ngFor. And don't use "this" in the html. Also I think you should have a getter on the form to be able to write : `control.activeMembers.value`

File: libs/organization/src/lib/components/organization-quorum-form/organization-quorum-form.component.html [hunk 6]
```diff
@@ -0,0 +1,22 @@
...
+      fxLayoutAlign="space-around center">
+      <mat-icon>
```
Comment (GrandSchtroumpf) [inline libs/organization/src/lib/components/organization-quorum-form/organization-quorum-form.component.html]: Put the mat-icon ouside of the span when ng-container is implemented

File: libs/organization/src/lib/organization.form.ts [hunk 7]
```diff
@@ -0,0 +1,38 @@
...
+    const org = createOrganization(organization);
+    super({
+      name: new FormControl(org.name),
+      actions: new FormArray(org.actions.map(action => new OrganizationActionForm(action)))
+    });
+  }
+  action(index: number): OrganizationActionForm {
+    return (<FormArray>this.get('actions')).at(index) as OrganizationActionForm;
+  }
+  patchValue(org: Organization) {
```
Comment (GrandSchtroumpf) [inline libs/organization/src/lib/organization.form.ts]: You should add the removeControl also.

File: libs/organization/src/lib/pages/organization-admin-view/organization-admin-view.component.ts [hunk 8]
```diff
@@ -14,36 +16,40 @@ export class OrganizationAdminViewComponent implements OnInit {
...
+  /** A number to iindicate which action we want to alter in the sidenav */
+  public activeForm: number;
+
   constructor(private query: OrganizationQuery, private service: OrganizationService) {}
 
   ngOnInit() {
-    this.organization$ = this.query.select('org');
+    this.organization$ = this.query
+      .select('org')
+      .pipe(
```
Comment (GrandSchtroumpf) [inline libs/organization/src/lib/pages/organization-admin-view/organization-admin-view.component.ts]: You have a problem of parenthesis here : 
```typescript
this.organization$ = this.query.select('org').pipe(
  tap(org => this.form = new OrganizationForm(org)),
  switchMap(org => this.form.valueChanges.pipe(startWith(org)))
);
```

File: libs/organization/src/lib/pages/organization-admin-view/organization-admin-view.component.ts [hunk 9]
```diff
@@ -14,36 +16,40 @@ export class OrganizationAdminViewComponent implements OnInit {
...
     this.opened = true;
-    this.optionsOrSigner = true;
-    this.action = action;
+    this.editContent = editContent;
+    this.activeForm = index;
   }
 
-  public deleteActiveSigner({ member, action }) {
-    this.service.deleteActiveSigner(member, action);
+  public deleteActiveSigner(uid: string) {
```
Comment (GrandSchtroumpf) [inline libs/organization/src/lib/pages/organization-admin-view/organization-admin-view.component.ts]: what to do with this function ?

File: libs/organization/src/lib/+state/organization.store.ts (hunk 2 above)
Comment (Maxstgt) [reply libs/organization/src/lib/+state/organization.store.ts]: We need this in the creation of organization process

File: libs/organization/src/lib/components/organization-action-item/organization-action-item.component.ts (hunk 4 above)
Comment (Maxstgt) [reply libs/organization/src/lib/components/organization-action-item/organization-action-item.component.ts]: Cause when I type it now, the mock value from pierre won't fit.

File: libs/organization/src/lib/organization.form.ts (hunk 7 above)
Comment (Maxstgt) [reply libs/organization/src/lib/organization.form.ts]: The question is how

File: libs/organization/src/lib/+state/organization.service.ts [hunk 10]
```diff
@@ -99,6 +99,7 @@ export class OrganizationService {
     const call = firebase.functions().httpsCallable('findOrgByName');
     return call({ prefix }).then(matchingOrganizations => matchingOrganizations.data);
   }
+<<<<<<< HEAD
```
Comment (Maxstgt) [inline libs/organization/src/lib/+state/organization.service.ts]: It is deleted in the following commit

File: libs/organization/src/lib/+state/organization.service.ts (hunk 1 above)
Comment (Maxstgt) [reply libs/organization/src/lib/+state/organization.service.ts]: It is removed. Dunno why your comment is still here :confused:

File: libs/organization/src/lib/components/organization-quorum-form/organization-quorum-form.component.ts [hunk 11]
```diff
@@ -0,0 +1,22 @@
...
+
+@Component({
+  selector: '[formGroup] activeMembers, organization-quorum-form',
+  templateUrl: './organization-quorum-form.component.html',
+  styleUrls: ['./organization-quorum-form.component.scss'],
+  changeDetection: ChangeDetectionStrategy.OnPush
+})
+export class OrganizationQuorumFormComponent {
+  @Output() deleteActiveSigner = new EventEmitter<number>();
+  @Output() newQuorumMember = new EventEmitter<number>();
```
Comment (GrandSchtroumpf) [inline libs/organization/src/lib/components/organization-quorum-form/organization-quorum-form.component.ts]: You don't need those output. You should update the control directly.

File: libs/organization/src/lib/organization.form.ts [hunk 12]
```diff
@@ -0,0 +1,32 @@
...
+  }
+}
+
+export class OrganizationForm extends FormEntity<Organization> {
+  constructor(organization: Partial<Organization> = {}) {
+    const org = createOrganization(organization);
+    super({
+      name: new FormControl(org.name),
+      address: new FormControl(org.address),
+      actions: FormList.factory(org.actions, (action) => new OrganizationActionForm(action))  // new FormArray(org.actions.map(action => ))
```
Comment (GrandSchtroumpf) [inline libs/organization/src/lib/organization.form.ts]: Remove comment at the end.

File: libs/utils/src/lib/form/forms/list.form.ts [hunk 13]
```diff
@@ -40,12 +61,30 @@ export class FormList<T> extends FormArray {
...
+      }
+    });
+    // If there is more value than form controls, remove it.
+    if (this.length > value.length) {
+      for (let i = value.length + 1; i++; i < this.length) {
+        this.removeAt(i);
+      }
+    }
   }
+  /* super.patchValue(value, options);} */
```
Comment (GrandSchtroumpf) [inline libs/utils/src/lib/form/forms/list.form.ts]: Remove this comment.

**Pull request merged**